src/main/base-proxy.ts

```diff
@@ -354,7 +354,7 @@ export class BaseProxy extends EventEmitter {
             const error = new ProxyConnectionFailed(upstream, connectRes.statusCode!);
             throw error;
         }
-        const connectionIdHeader = String(connectRes.headers['x-connection-id'] || '');
+        const connectionIdHeader = String(connectRes.headers['x-connection-id'] || inboundConnectReq.headers['x-partition-id'] || '');
```

Comment:
To be fair, the fix works *without* this change. However, when an `x-connection-id` header does not exist, for whatever reason (e.g. the upstream is a non-uniproxy proxy), we should still honour the use-partition-as-connection-id logic.

This is applicable only to using non-uniproxy upstreams, so it is not testable right now.